src/tilemap_layer.cpp

```diff
@@ -186,26 +218,35 @@ void TilemapLayer::Draw(Bitmap& dst, int z_order) {
 		++tiles_y;
 	}
 
-	for (int x = 0; x < tiles_x; x++) {
-		for (int y = 0; y < tiles_y; y++) {
+	const bool loop_h = Game_Map::LoopHorizontal();
+	const bool loop_v = Game_Map::LoopVertical();
+
+	auto div_rounding_down = [](int n, int m) {
+		if (n >= 0) return n / m;
+		return (n - m + 1) / m;
+	};
+	auto mod = [](int n, int m) {
```

Comment:
For this we already have Utils::PositiveModulo.
But my benchmarks say that your version is faster than what we already have, how about replacing it?

Reply:
This isn't my version, it was already here. Given this has a branch, I would expect either the utils variant to be faster or the compiler to be smart enough to optimize them both to the same thing.

If you saw utils version to be slower, it's probably because it's currently not inlined.

Reply:
https://gcc.godbolt.org/z/rFUMwH

They do produce different code. You trade a branch for 2 divisions.

I would want to benchmark this more carefully before deciding which to use.

Reply:
Punting this one to #2042